app_dart/lib/src/request_handlers/check_flaky_tests_and_update_github.dart

```diff
@@ -0,0 +1,450 @@
+      if (match != null) {
+        nameToExistingPRs[match.namedGroup('name')] = ExistingGithubPR(
+          match.namedGroup('name'),
+          pr
+        );
+      }
+    }
+    return nameToExistingPRs;
+  }
+
+  Set<String> _getImportantFlakes(List<BuilderStats> statsList, double threshold) {
+    final Set<String> importantFlakes = <String>{};
+    for (final BuilderStats stats in statsList) {
+      if (stats.flakyRate > threshold) {
+        importantFlakes.add(stats.name);
+      }
+    }
+    if (importantFlakes.isNotEmpty) {
+      return importantFlakes;
+    }
+    BuilderStats mostImportant;
+    for (final BuilderStats stats in statsList) {
+      if (mostImportant == null || mostImportant.flakyRate < stats.flakyRate) {
+        mostImportant = stats;
+      }
+    }
+    return <String>{
+      if (mostImportant != null)
+        mostImportant.name,
+    };
+  }
+
+  Future<void> _updateFlakes({
+    GitHub client,
+    BuilderStats newStats,
+    Map<String, ExistingGithubIssue> existingIssues,
+    Map<String, ExistingGithubPR> existingPRs,
+    bool isImportant = false,
+  }) async {
+    // Don't create a new issue if there is a recent closed issue within
+    // _kGracePeriodForClosedFlake days. It takes time for the flaky ratio to go
+    // down after the fixed is merged.
+    Issue issue;
+    if (!existingIssues.containsKey(newStats.name) ||
+        (existingIssues[newStats.name].issue.state == 'closed' &&
+         existingIssues[newStats.name].issue.closedAt.difference(DateTime.now())> const Duration(days: _kGracePeriodForClosedFlake))) {
+      if (!isImportant) {
+        throw 'This handler should only create new issue for important flake, something went wrong!';
+      }
+      issue = await _fileNewIssue(
+        stats: newStats,
+        client: client,
+      );
+    } else if(existingIssues.containsKey(newStats.name) && existingIssues[newStats.name].issue.state == 'open') {
+      issue = await _updateIssue(
+        stats: newStats,
+        existingIssue: existingIssues[newStats.name],
+        client: client,
+        isImportant: isImportant
+      );
+    } else {
+      // Do nothing if there is a closed issue within _kGracePeriodForClosedFlake days.
+      return;
+    }
+    if (isImportant) {
+      if (issue == null) {
+        throw 'This issue not be null for important flake, something went wrong!';
+      }
+      await _fileNewPullRequestIfNeeded(
+          stats:newStats,
+          issue: issue,
+          existingPRs: existingPRs,
+          client: client,
+      );
+    }
+  }
+
+  Future<void> _maybeCloseIssue(Issue issue, GitHub client) async {
+    if (issue.assignee != null) {
+      return;
+    }
+    await client.issues.createComment(config.flutterSlug, issue.number, kCloseIssueComment);
+    await client.issues.edit(
+      config.flutterSlug,
+      issue.number,
+      IssueRequest(
+        state: 'closed',
```

Comment:
Does it make sense if we leave the comment to suggest a fix is validated, and create a PR to mark the test as unflaky?
As for closing the issue, it maybe good to leave the decision to the bug owner?